src/instance.ts

```diff
@@ -192,6 +238,254 @@ class Instance extends common.GrpcServiceObject {
+        ? queryOrCallback
+        : ({} as ListBackupOperationsRequest);
+
+    const reqOpts = extend({}, query, {
+      parent: this.formattedName_,
+    });
+    this.request<IBackup[]>(
+      {
+        client: 'DatabaseAdminClient',
+        method: 'listBackupOperations',
+        reqOpts,
+        gaxOpts: query,
+      },
+      (err, operations, ...args) => {
+        callback!(err, operations, ...args);
+      }
+    );
+  }
+
+  /**
+   * Query object for listing database operations.
+   *
+   * @typedef {object} ListDatabaseOperationsRequest
+   * @property {boolean} [autoPaginate=true] Have pagination handled
+   *     automatically.
+   * @property {number} [maxApiCalls] Maximum number of API calls to make.
+   * @property {number} [maxResults] Maximum number of items to return.
+   * @property {number} [pageSize] Maximum number of results per page.
+   * @property {string} [pageToken] A previously-returned page token
+   *     representing part of the larger set of results to view.
+   */
+  /**
+   * @typedef {array} ListDatabaseOperationsResponse
+   * @property {IOperation[]} 0 Array of {@link IOperation} instances.
+   * @property {object} 1 The full API response.
+   */
+  /**
+   * List pending and completed operations for all databases in the instance.
+   *
+   * @see {@link Database.listDatabaseOperations}
+   *
+   * @param query query object for listing database operations.
+   * @returns {Promise<ListDatabaseOperationsResponse>} when resolved, contains a paged list of database operations.
+   *
+   * @example
+   * const {Spanner} = require('@google-cloud/spanner');
+   * const spanner = new Spanner();
+   * const instance = spanner.instance('my-instance');
+   * const [operations] = await instance.listDatabaseOperations();
+   * // ... then do something with the operations
+   */
+  listDatabaseOperations(
+    query?: ListDatabaseOperationsRequest
+  ): Promise<ListDatabaseOperationsResponse>;
+  listDatabaseOperations(callback: ListDatabaseOperationsCallback): void;
+  listDatabaseOperations(
+    query: ListDatabaseOperationsRequest,
+    callback: ListDatabaseOperationsCallback
+  ): void;
+
+  listDatabaseOperations(
+    queryOrCallback?:
+      | ListDatabaseOperationsRequest
+      | ListDatabaseOperationsCallback,
+    cb?: ListDatabaseOperationsCallback
+  ): void | Promise<ListDatabaseOperationsResponse> {
+    const callback =
+      typeof queryOrCallback === 'function' ? queryOrCallback : cb!;
+    const query =
+      typeof queryOrCallback === 'object'
+        ? queryOrCallback
+        : ({} as ListDatabaseOperationsRequest);
+
+    const reqOpts = extend({}, query, {
+      parent: this.formattedName_,
+    });
+    this.request<IBackup[]>(
+      {
+        client: 'DatabaseAdminClient',
+        method: 'listDatabaseOperations',
+        reqOpts,
+        gaxOpts: query,
+      },
+      (err, operations, ...args) => {
+        callback!(err, operations, ...args);
```

Comment:
Is `operations` just an array of operation descriptive objects? If so, we should return them in our custom `Operation` objects here: https://github.com/googleapis/nodejs-spanner/blob/810aa2aebdfba8a885c618ab035fd3b6b579dfe5/src/index.ts#L751.

Reply:
I've tried to get this working with `GrpcOperation` objects as suggested, but it seems at least with my setup that the `getMetadata()` call always fails on them and I'm not sure why.

https://github.com/prunge-helix/nodejs-spanner/commit/a9034d1af17ff55cc81cfbf686cdb68ac677ac87

I've put what I have in a new branch 'prunge/backups-samples-2020-operations' - @skuruppu hopefully that provides a starting point.

Reply:
This is the error I got:

```
(node:21311) UnhandledPromiseRejectionWarning: TypeError: gapic.v1[clientName] is not a constructor
    at auth.getProjectId (/home/prunge/data/helix/nodejs-spanner/build/src/index.js:713:47)
    at getProjectIdAsync.then.r (/home/prunge/data/helix/nodejs-spanner/node_modules/google-auth-library/build/src/auth/googleauth.js:65:48)
(node:21311) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
```

Reply:
Thank you for trying that. It looks like our custom `spanner.operation()` method doesn't work anymore, and the `operations` argument from `this.request<IOperation[]>()` is already a custom Operation object from gax. That object doesn't have a `getMetadata()` method, though. They call it `getOperation()`.

Reply:
Thanks for the commit of your try @prunge-helix. I couldn't find a way to get this working without a lot of changes IMHO.

1. The service method config in `src/common-grpc/operation.ts` is wrong (it was probably correct some time ago but it doesn't match how services are configured now).
2. Even if I fix the config, `src/v1/` doesn't include the service so there's no way to initialize the Operations client in `src/index.ts`.

I'm not sure how to fix this.

The other thing is that `spanner.operation()` is not used anywhere else in the repo (other than in unit tests for GrpcOperation).

@stephenplusplus I'm not sure that having this operation type adds any value so I would propose to remove it and leave the implementation as is. But let me know if you think otherwise.

Please also note that the whole feature launch is blocked on this PR. So I would really appreciate a resolution quickly either way:
1. If there's a fix that you can help me with, I'd appreciate it.
2. Delete `spanner.operation()` and `src/common-grpc/operation.ts` to avoid confusion in the future (in a separate PR of course).

Reply:
I spoke to @alexander-fenster offline and he said that clients should be using the Gax operations directly. In light of that, I've created #879 to delete `spanner.operation()`.

Reply:
Resolving this comment for now but I'll be happy to update the implementation once the GAPIC generator can support accessing operations by their name.